llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -1297,10 +1295,9 @@ void AMDGPUTargetLowering::analyzeFormalArgumentsCompute(
 
       unsigned PartOffset = 0;
       for (unsigned i = 0; i != NumRegs; ++i) {
-        State.addLoc(CCValAssign::getCustomMem(InIndex++, RegisterVT,
-                                               BasePartOffset + PartOffset,
-                                               MemVT.getSimpleVT(),
-                                               CCValAssign::Full));
+        State.addLoc(CCValAssign::getCustomMem(
+            Arg.getArgNo(), RegisterVT, BasePartOffset + PartOffset,
```

Comment:
Was ValNo just not used before and you have co-opt it to mean the original argument idx, or am I misunderstanding this change?

Reply:
Yes. `ValNo` is not used in the AMD backend and the docs in `CCValAssign` suggest it should be an argument number.